src/deadline/job_attachments/api/manifest.py

```diff
@@ -0,0 +1,347 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+import datetime
+from io import BytesIO
+from pathlib import Path
+from typing import Any, List, Optional
+
+import boto3
+from botocore.client import BaseClient
+
+from deadline.client import api
+from deadline.client.api._session import _get_queue_user_boto3_session, get_default_client_config
+from deadline.client.cli._common import _ProgressBarCallbackManager
+from deadline.client.cli._groups.click_logger import ClickLogger
+from deadline.client.config import config_file
+from deadline.job_attachments._aws.aws_clients import get_s3_client, get_s3_transfer_manager
+from deadline.job_attachments._diff import compare_manifest, pretty_print_cli
+from deadline.job_attachments._glob import _process_glob_inputs
+from deadline.job_attachments._utils import _glob_paths
+from deadline.job_attachments.asset_manifests.base_manifest import (
+    BaseAssetManifest,
+    BaseManifestPath,
+)
+from deadline.job_attachments.asset_manifests.decode import decode_manifest
+from deadline.job_attachments.caches.hash_cache import HashCache
+from deadline.job_attachments.download import download_file_with_s3_key
+from deadline.job_attachments.models import (
+    S3_MANIFEST_FOLDER_NAME,
+    GlobConfig,
+    JobAttachmentS3Settings,
+    ManifestDiff,
+    ManifestDownload,
+    ManifestDownloadResponse,
+    ManifestSnapshot,
+)
+from deadline.job_attachments.upload import FileStatus, S3AssetManager, S3AssetUploader
+
+"""
+APIs here should be business logic only. It should perform one thing, and one thing well. 
+It should use basic primitives like S3 upload, download, boto3. 
+These APIs should be boto3 session agnostic and given which underlying credentials are used.
+"""
+
+
+def _manifest_upload(
```

Comment:
As far as I understand, the APIs here are serving python API users. The function here should be public no? We probably don't need the `_` prefix.